sympy/solvers/ode/systems.py

```diff
@@ -349,6 +450,120 @@ def _matrix_is_constant(M, t):
+        f(x1, y1, z1, ..., x, y, z, ..., t) = g(x1, y1, z1, ..., x, y, z, ..., t)
+
+    Where $f$ and $g$ are list of expressions in terms of $x1$, $y1$, $z1$, ... which
+    are the first order derivatives of depedent variables $x$, $y$, $z$, ... from the
+    list *funcs* dependent on the independent variable $t$.
+
+    This function takes the above system of equations and converts it into:
+
+    .. math::
+        x1 = f1(x, y, z, ..., t)
+        y1 = f2(x, y, z, ..., t)
+        z1 = f3(x, y, z, ..., t)
+        ...
+
+    Where $f1$, $f2$, $f3$, ... are functions dependent on $x$, $y$, $z$, ..., $t$.
+
+    The above converted system of equations is then returned by this function.
+
+    Parameters
+    ==========
+
+    eqs: List
+        List of ODEs
+    funcs: List
+        List of dependent variables
+    t: Symbol
+        Independent variable of the equations in eqs
+
+    Raises
+    ======
+
+    ODENonLinearError
+        When the system of equations is nonlinear
+    ODEOrderError
+        When the system of equations have an order greater than 1
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, Eq, Function
+    >>> from sympy.solvers.ode.systems import canonical_equations
+    >>> from sympy.abc import t
+
+    >>> x, y = symbols("x y", cls=Function)
+    >>> x1 = x(t).diff(t)
+    >>> y1 = y(t).diff(t)
+
+    Now, we will look at a system of ODEs which is not in canonical form:
+
+    >>> eqs = [Eq(2*x1 + y1, x(t) + t), Eq(y1, y(t) + x1)]
+
+    The canonical form of the equation is given as:
+
+    >>> canonical_equations(eqs, [x(t), y(t)], t)
+    [Eq(Derivative(x(t), t), t/3 + x(t)/3 - y(t)/3),
+     Eq(Derivative(y(t), t), t/3 + x(t)/3 + 2*y(t)/3)]
+
+
+    The function raises an ODEOrderError since it only deals with
+    system of ODEs in their first order.
+
+    >>> eqs = [Eq(x1.diff(t), y(t)), Eq(y1, x(t))]
+
+    >>> canonical_equations(eqs, [x(t), y(t)], t)
+    Traceback (most recent call last):
+    ...
+    ODEOrderError: Cannot represent system in 1-order canonical form
+
+
+    This function raises an ODENonLinearError when one of the first order
+    terms is nonlinear.
+
+    >>> eqs = [Eq(x1**2, x(t) + y(t) + t), Eq(y1 + x1, x(t) + 2*y(t))]
+
+    >>> canonical_equations(eqs, [x(t), y(t)], t)
+    Traceback (most recent call last):
+    ...
+    ODENonLinearError: System of ODEs is nonlinear
+
+    See Also
+    ========
+
+    solve: Solves for the first order dependent variables
+
+    """
+    from sympy.solvers.solvers import solve
+
+    # For now the system of ODEs dealt by this function can have a
+    # maximum order of 1.
+    if any(ode_order(eq, func) > 1 for eq in eqs for func in funcs):
+        msg = "Cannot represent system in {}-order canonical form"
+        raise ODEOrderError(msg.format(1))
+
+    canon_eqs = solve(eqs, *[func.diff(t) for func in funcs], dict=True)
+
+    if len(canon_eqs) != 1:
+        raise ODENonLinearError("System of ODEs is nonlinear")
```

Comment:
For what we are doing here I think it would be better to return the multiple solutions so that they can be solved. It's possible that the ODEs are nonlinear but that solving leads to multiple linear systems e.g.:
```julia
In [48]: x, y = symbols('x, y', cls=Function)                                                                                     

In [49]: t = Symbol('t')                                                                                                          

In [50]: sys = [Eq(x(t).diff(t)**2, y(t)**2), Eq(y(t).diff(t)**2, x(t)**2)]                                                       

In [51]: solve(sys, [x(t).diff(t), y(t).diff(t)], dict=True)                                                                      
Out[51]: 
⎡⎧d                d              ⎫  ⎧d                d             ⎫  ⎧d               d              ⎫  ⎧d               d   
⎢⎨──(x(t)): -y(t), ──(y(t)): -x(t)⎬, ⎨──(x(t)): -y(t), ──(y(t)): x(t)⎬, ⎨──(x(t)): y(t), ──(y(t)): -x(t)⎬, ⎨──(x(t)): y(t), ──(y
⎣⎩dt               dt             ⎭  ⎩dt               dt            ⎭  ⎩dt              dt             ⎭  ⎩dt              dt  

          ⎫⎤
(t)): x(t)⎬⎥
          ⎭⎦
```
For our purposes here this function seems unnecessary as it is basically just a call to `solve`. I think a function *like* this can be of more general use because you would need to put the ODEs into this form in order to solve them numerically with lambdify and e.g. scipy.odeint  but more thought is needed to make that work fully. I suggest to mark this function as private/internal by calling it `_canonical_equations` for now. Then the signature could be changed later if it seems useful.

Reply:
@oscarbenjamin So, I would mark this function as private and just let it be for now, or should I update this function right now to handle multiple linear systems obtained from the solutions?